docs/engineering/dev-process.md

```diff
@@ -0,0 +1,50 @@
+---
+id: dev-process
+title: Development Process
+sidebar_label: Development Process
+---
+
+
+
+## Releasing Code
+0. A release owner is delegated to follow the process below and work with the
+   team to tie up any loose ends
+0. The pre-flight checklist:
+    0. Ensure there are no critical patches for this tag that haven't landed
+       yet
+    0. Fill out a new section in the [deployment doc][deployment-doc] for the Train you are tagging
+    0. Ensure you've got appropriate qa signoffs
+    0. Ensure you don't have any modified files or code laying around before
+       you start the tag
+0. Run [release.sh][release.sh] from the root of the repository.  Make sure
```

Comment:
Not sure if it makes sense to mention point releases here (maybe that is worth its own section). Basically it is same process except you run with `release.sh patch`, while in the train branch.

Reply:
I added an FAQ section at the bottom, although if you want to expand, please do